Shared/Samples/Navigate route with rerouting/NavigateRouteWithReroutingView.Model.swift

```diff
@@ -0,0 +1,304 @@
+        /// - Parameter status: The `TrackingStatus`.
+        func updateGraphics(using status: TrackingStatus) async {
+            // Update the route graphics.
+            remainingRouteGraphic.geometry = status.routeProgress.remainingGeometry
+            
+            if let currentPosition = locationDisplay.location?.position {
+                traversedRouteBuilder.add(currentPosition)
+                traversedRouteGraphic.geometry = traversedRouteBuilder.toGeometry()
+            }
+            
+            // Update the status message.
+            switch status.destinationStatus {
+            case .approaching, .notReached:
+                // Format the route's remaining distance and time.
+                let distanceRemainingText = status.routeProgress.remainingDistance.distance.formatted()
+                
+                let dateInterval = DateInterval(start: .now, duration: status.routeProgress.remainingTime)
+                let dateRange = dateInterval.start..<dateInterval.end
+                let timeRemainingText = dateRange.formatted(
+                    .components(style: .abbreviated, fields: [.day, .hour, .minute, .second])
+                )
+                
+                statusMessage = """
+                Distance remaining: \(distanceRemainingText)
+                Time remaining: \(timeRemainingText)
+                """
+                
+                // Get the next direction from the route's direction maneuvers.
+                let nextManeuverIndex = status.currentManeuverIndex + 1
+                if route.directionManeuvers.indices.contains(nextManeuverIndex) {
+                    let nextDirection = route.directionManeuvers[nextManeuverIndex].text
+                    statusMessage.append("\nNext direction: \(nextDirection)")
+                }
+                
+            case .reached:
+                if status.remainingDestinationCount > 1 {
+                    statusMessage = "Intermediate stop reached, continue to next stop."
+                    try? await routeTracker.switchToNextDestination()
+                } else {
+                    await locationDisplay.dataSource.stop()
+                    statusMessage = "Destination reached."
+                }
+                
+            @unknown default:
+                break
+            }
+        }
+        
+        /// Speaks a given voice guidance.
+        /// - Parameter voiceGuidance: The `VoiceGuidance`.
+        func speakVoiceGuidance(_ voiceGuidance: VoiceGuidance) {
+            guard !voiceGuidance.text.isEmpty else { return }
+            
+            let utterance = AVSpeechUtterance(string: voiceGuidance.text)
+            speechSynthesizer.stopSpeaking(at: .word)
+            speechSynthesizer.speak(utterance)
+        }
+        
+        /// Initializes the location display and route graphic.
+        private func initializeGraphics() {
```

Comment:
I don't think `initializeGraphics` describes everything this function does. It initializes the viewpoint, which isn't a graphic. Maybe something like tracking progress instead of graphics.